consul/catalog_endpoint.go

```diff
@@ -53,11 +53,14 @@ func (c *Catalog) Register(args *structs.RegisterRequest, reply *struct{}) error
 	}
 
 	if args.Check != nil {
-		if args.Check.CheckID == "" && args.Check.Name != "" {
-			args.Check.CheckID = args.Check.Name
+		args.Checks = append(args.Checks, args.Check)
```

Comment:
after we append to Checks, we should probably nil out `args.Check` so that the FSM doesn't double create.